### DISCUSSION TITLE
How to create input without using a parser?

### DISCUSSION THREAD

**Discussion by 9527567** [alchemistry/alchemlyb#401] [Q&A] [answered]
In this [issue](https://github.com/Gallicchio-Lab/AToM-OpenMM/issues/87), with @egallicc's help, using [pymbar](https://github.com/choderalab/pymbar) to analyze the binding energy of [AToM-OpenMM](https://github.com/Gallicchio-Lab/AToM-OpenMM) has become a reality. I tried to read and understand the [tutorial](https://alchemlyb.readthedocs.io/en/latest/parsing.html), but unfortunately I am not an expert in statistics.

**Comment by orbeckst** [comment]
I converted your issue to a discussion because it's not an issue in the library but a question.

If you already have data in a form suitable for pymbar then you might not need alchemlyb.

If you want to use alchemlyb then you have to bring your data into the [standard form of raw data](https://alchemlyb.readthedocs.io/en/latest/parsing.html#standard-forms-of-raw-data) as described in the docs; the [u_nk standard form](https://alchemlyb.readthedocs.io/en/latest/parsing.html#u-nk-standard-form) seems to be most relevant to you, given that you want to use MBAR. Once you have a pandas data frame with the correct information, you can use alchemlyb's subsampler, estimator, and visualization modules.

**Reply by xiki-tempula** [reply, marked as answer]
If you have the reduced potential as a numpy array. I have this old script that might help you.
https://github.com/alchemistry/alchemlyb/discussions/296#discussioncomment-5120835

**Reply by 9527567** [reply]
thank you.
